### ISSUE TITLE
Extension not working through the api

### ISSUE THREAD

**Issue by iChristGit** [oobabooga/text-generation-webui#4565]
I run the webui with --api (now with --extension openai)
And SillyTavern conencts to it.
When using the extension web_search (https://github.com/simbake/web_search)
I can get good informaion with search on oobabooga, but on SillyTavern it just ignores the "search" activation word.
Can I somehow make it work? so the extension works with other things that rely on the api?

Thanks!

**Comment by yhyu13** [comment]
So you are using SillyTavern as the client side instead of textgen's gradio webui? And then use openai api to connect with SIllyTavern

the web search extension meant to work inside textgen's gradio webui as extension don't know each other. But I believe you just need to rewire the openai chat completion backend from pure textgen completion, to textgen completion with websearch extension

**Comment by yhyu13** [comment]
![Code_lGuQgLMfVD](https://github.com/oobabooga/text-generation-webui/assets/19365678/11518ac8-1a21-4c8c-83b6-97f6566ef1ea)

I don't have an available isntance at this moment, but I found this line of code suspicious and worth debugging. The apply_extension("input") is where the search web extension modified input's "search" activation word.

Also, you model loader would affect whether or not the genearte_reply is hf or custom, which treat extension modified input differently

**Comment by simbake** [comment]
@yhyu13 Hi, how do I connect websearch with the openai api in ooba?

**Comment by yhyu13** [comment]
I don't think that web search extention works. It actually does not retrive anything for me. You'd better use a one that uses serp api instead of web scraping.

To figure out how the websearch would work together with openai-api, we should debug a little bit.

The Oob extension works by modifying prompt input by each extension. So we need somehow make the websearch extension modify the prompt and add its search result to prompt before sending it to openai-api extension.

**Comment by bitsnaps** [comment]
> I don't think that web search extention works. It actually does not retrive anything for me. You'd better use a one that uses serp api instead of web scraping.

This should probably be implemented using different options/tags (e.g. `--web_search [serp|scrape...]`)
